apptest/deployer/schema.yaml

```diff
@@ -1,3 +1,62 @@
+x-google-marketplace:
+  schemaVersion: v2
+
+  # Must match the version of the Application custom resource object.
+  applicationApiVersion: v1beta1
+
+  # The release version must match the release tag on the deployer
+  publishedVersion: '1.4.0'
+
+  images:
+    postgres:
+      properties:
+        imageRepo:
+          type: REPO_WITH_REGISTRY
+        imageTag:
+          type: TAG
+    nginx:
+      properties:
+        imageRepo:
+          type: REPO_WITH_REGISTRY
+        imageTag:
+          type: TAG
+    tester:
+      properties:
+        imageRepo:
+          type: REPO_WITH_REGISTRY
+        imageTag:
+          type: TAG
+    init:
+      properties:
+        imageInitFull:
+          type: FULL
+        imageInitRegistry:
+          type: REGISTRY
+        imageInitRepo:
+          type: REPO_WITHOUT_REGISTRY
+        imageInitTag:
+          type: TAG
+
+  clusterConstraints:
+    resources:
+    - requests:
+        cpu: 1000m
+        memory: 3750Mi
+      affinity:
+        simpleNodeAffinity:
+          type: REQUIRE_MINIMUM_NODE_COUNT
+          minimumNodeCount: 3
+
+  clusterConstraints:
+    resources:
+    - requests:
+        cpu: 1000m
```

Comment:
Is this param required? I think we may want more cores for a deployment.

Reply:
The resource requests specifies the minimum amount that the marketplace containers need to run (i.e. the amount that kubelet needs to guarantee to a pod when the pod is scheduled on a node) . (Ref: https://github.com/kubernetes/community/blob/master/contributors/design-proposals/node/resource-qos.md#introduction).

If we don't add this request setting, it means that someone may try to click-and-run our marketplace app with insufficient cores.

We could potentially raise this minimum value, but it means that we wouldn't allow customers to click-and-run our app unless they're running it on a GKE cluster with that number of cores.

I know the app will work with the 1000m (1 VCPU) value, since I took it from the GKE cluster that we use to test our app in CI: https://console.cloud.google.com/kubernetes/list?authuser=1&folder=&organizationId=&walkthrough_tutorial_id=java_gae_quickstart&project=conjur-cloud-launcher-onboard .

WDYT?

Reply:
LGTM